activerecord/test/cases/finder_test.rb

```diff
@@ -987,8 +1001,9 @@ def test_find_last_with_offset
     devs = Developer.order('id')
 
     assert_equal devs[2], Developer.offset(2).first
-    assert_equal devs[-3], Developer.offset(2).last
-    assert_equal devs[-3], Developer.offset(2).last
+    assert_equal devs[-1], Developer.offset(2).last # note that this behavior is deprecated and will raise an exception in Rails 5.1
```

Comment:
I didn't even realize that `assert_deprecated` existed! That's great. Change made in 53edb98.